File: src/main/java/com/iobestgroup/donkeymoney/user/DMUser.java
package com.iobestgroup.donkeymoney.user;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.hibernate.validator.constraints.Email;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity
@Table(name = "users")
public class DMUser implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "last_name")
    private String lastName;

    @Email
    @Column(name = "email", unique = true)
    private String email;

    @Column(name = "password")
    private String password;

    @JsonIgnore
    public String getPassword() {
        return password;
    }

    @JsonProperty
    public void setPassword(String password) {
        this.password = password;
    }


}

File: src/main/java/com/iobestgroup/donkeymoney/user/DMUser.kt
package com.iobestgroup.donkeymoney.user

import org.hibernate.validator.constraints.Email

import javax.persistence.*

@Entity
@Table(name = "users")
data class DMUser(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null,


    @Email
    @Column(name = "username", unique = true)
    var email: String? = null,


    @Column(name = "password")
    var password: String = "",


    @Column(name = "securityToken")
    var securityToken: String = ""
)

